### ISSUE TITLE
Bug: [consistent-type-imports] Incorrectly suggests `import type` on import assertion

### ISSUE THREAD

**Issue by somebody1234** [typescript-eslint/typescript-eslint#7527]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&showAST=es&fileType=.tsx&code=JYWwDg9gTgLgBAKjgQwM4oFKogOzgMyghDgHIA6AemXICtsdSVVUBTWOAbzhgE8xWALjL1cTAL4AoSQGNcqeAHcAFshjDkWXOQCCAbiA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWpQC2xDtHyoMkRNGhTI4MAF8QaoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```typescript
import * as aJson from './a.json' assert { type: 'json' }

const what: aJson.A;
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/consistent-type-imports": "error"
  }
}
```

### Expected Result

No error

### Actual Result

Error. When fixing it, TypeScript says `Import assertions cannot be used with type-only imports or exports. ts(2822)`.

Importantly, the autofix "fixes" it into invalid TypeScript code.

**Comment by rielAsh24** [comment]
Hey @JoshuaKGoldberg @somebody1234 . I'd be glad to look into this!

**Comment by rielAsh24** [comment]
A quick question; I wish to test the `repro code` against my changes. Do I need to add another test or does passing the tests in the codebase suffice?

**Comment by bradzacher** [comment]
With any change we require new tests to ensure the new code is working as expected!

**Comment by rielAsh24** [comment]
> With any change we require new tests to ensure the new code is working as expected!

I have already added a test case in the pull request, so if there's more required, I can surely work on it.
